pkg/scheduler/server/internal/etcd/config.go

```diff
@@ -39,20 +42,38 @@ func config(opts Options) (*embed.Config, error) {
 	config.ExperimentalWarningApplyDuration = time.Second * 5
 
 	if opts.Security.MTLSEnabled() {
-		hostName := "dapr-scheduler-server." + opts.Security.ControlPlaneNamespace() + ".svc"
 		info := transport.TLSInfo{
 			ClientCertAuth:      true,
 			InsecureSkipVerify:  false,
 			SkipClientSANVerify: false,
-			AllowedHostname:     hostName,
-			EmptyCN:             true,
-			CertFile:            filepath.Join(*opts.Security.IdentityDir(), "cert.pem"),
-			KeyFile:             filepath.Join(*opts.Security.IdentityDir(), "key.pem"),
-			ClientCertFile:      filepath.Join(*opts.Security.IdentityDir(), "cert.pem"),
-			ClientKeyFile:       filepath.Join(*opts.Security.IdentityDir(), "key.pem"),
-			TrustedCAFile:       filepath.Join(*opts.Security.IdentityDir(), "ca.pem"),
-			ServerName:          hostName,
+			AllowedHostnames: []string{
+				fmt.Sprintf("dapr-scheduler-server-0.dapr-scheduler-server.%s.svc", opts.Security.ControlPlaneNamespace()),
+				fmt.Sprintf("dapr-scheduler-server-1.dapr-scheduler-server.%s.svc", opts.Security.ControlPlaneNamespace()),
+				fmt.Sprintf("dapr-scheduler-server-2.dapr-scheduler-server.%s.svc", opts.Security.ControlPlaneNamespace()),
+			},
+			EmptyCN:        true,
+			CertFile:       filepath.Join(*opts.Security.IdentityDir(), "cert.pem"),
+			KeyFile:        filepath.Join(*opts.Security.IdentityDir(), "key.pem"),
+			ClientCertFile: filepath.Join(*opts.Security.IdentityDir(), "cert.pem"),
+			ClientKeyFile:  filepath.Join(*opts.Security.IdentityDir(), "key.pem"),
+			TrustedCAFile:  filepath.Join(*opts.Security.IdentityDir(), "ca.pem"),
+			ServerName:     fmt.Sprintf("%s.dapr-scheduler-server.%s.svc", opts.Name, opts.Security.ControlPlaneNamespace()),
 		}
+
+		b, err := os.ReadFile(filepath.Join(*opts.Security.IdentityDir(), "cert.pem"))
+		if err != nil {
+			return nil, err
+		}
+
+		certs, err := pem.DecodePEMCertificatesChain(b)
+		if err != nil {
+			return nil, err
+		}
+
+		if !slices.Contains(certs[0].DNSNames, fmt.Sprintf("*.dapr-scheduler-server.%s.svc", opts.Security.ControlPlaneNamespace())) {
```

Comment:
I tested locally and am seeing this logged from etcd:
```
{"level":"warn","ts":"2025-02-11T19:28:42.785510Z","caller":"embed/config_logging.go:161","msg":"rejected connection on peer endpoint","remote-addr":"10.244.1.12:44178","server-name":"dapr-scheduler-server-1.dapr-scheduler-server.dapr-tests.svc","ip-addresses":[],"dns-names":["*.dapr-scheduler-server.dapr-tests.svc"],"error":"tls: \"10.244.1.12\" does not match any of DNSNames [\"*.dapr-scheduler-server.dapr-tests.svc\"]"}
```
I dont think etcd likes the wildcard